staging/src/k8s.io/client-go/rest/client.go

```diff
@@ -244,7 +244,9 @@ func (c *RESTClient) Patch(pt types.PatchType) *Request {
 
 // Get begins a GET request. Short for c.Verb("GET").
 func (c *RESTClient) Get() *Request {
-	return c.Verb("GET")
+	// Default cache-control to "no-store, no-cache" to make sure we
+	// don't store sensitive information.
+	return c.Verb("GET").SetHeader("Cache-control", "no-store, no-cache")
```

Comment:
@liggitt To clarify, is it that you don't think the `Get` function should be defining the caching decision, or that setting the header is not the correct way to define it?